bftengine/src/bftengine/PersistentStorageImp.hpp

```diff
@@ -92,14 +92,15 @@ const uint16_t numOfLastExitFromViewDescObjs = kWorkWindowSize + 1;
 enum DescMetadataParameterIds {
   LAST_EXIT_FROM_VIEW_DESC = WIN_PARAMETERS_NUM + reservedWindowParamsNum,
   LAST_EXEC_DESC = LAST_EXIT_FROM_VIEW_DESC + numOfLastExitFromViewDescObjs,
+  LAST_STABLE_CHECKPOINT_DESC,
```

Comment:
Are you going to save/retrieve all stable checkpoints at once without having the ability to update/get them for a specific replica only?

Reply:
Yes, we need all the quorum of CheckpointMsg-s that serve as proof for our latest Stable Sequence Number. We store them in the transaction that we open in **onSeqNumIsStable** and in case the Replica has been restarted we recover them from persistent storage and insert all of them in **checkpointsLog**. Once the Replica moves to higher Sequence Numbers and collects Stable Checkpoint it will call **advanceActiveWindow** to clean up the old CheckpointMsg-s and will persist the new ones.